client/my-sites/checkout/controller.jsx

```diff
@@ -102,13 +102,28 @@ export function checkout( context, next ) {
 		( !! jetpackPurchaseToken || !! jetpackPurchaseNonce );
 	const jetpackSiteSlug = context.params.siteSlug;
 
+	const isGiftPurchase = context.pathname.includes( '/gift/' );
+
 	// Do not use Jetpack checkout for Jetpack Anti Spam
 	if ( 'jetpack_anti_spam' === context.params.productSlug ) {
 		page( context.path.replace( '/checkout/jetpack', '/checkout' ) );
 		return;
 	}
 
-	if ( ! selectedSite && ! isDisallowedForSitePicker && ! isJetpackCheckout ) {
+	const shouldAllowNoSelectedSite = () => {
+		if ( isDisallowedForSitePicker ) {
+			return true;
+		}
+		if ( isJetpackCheckout ) {
+			return true;
+		}
+		if ( isGiftPurchase ) {
+			return true;
+		}
+		return false;
+	};
```

Comment:
What's the reason for this to be a function?

Reply:
My intent here was to make the logic more readable. I think that by using early returns in a function, the conditions of this variable are pretty clear, whereas trying to put them into a single condition would have been less so.

I found that this statement is more readable:

```js
if ( ! selectedSite && ! shouldAllowNoSelectedSite() )
```

...and that this is less readable:

```js
if ( ! selectedSite && ! isDisallowedForSitePicker && ! isJetpackCheckout && ! isGiftPurchase )
```

...but I'm open to disagreement!